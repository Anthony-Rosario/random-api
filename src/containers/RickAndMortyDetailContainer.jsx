import React, { Component } from 'react'
import CharacterDetail from '../components/app/characters/CharacterDetail'
import { findCharacterById } from '../services/rickandmortyApi'

export default class RickAndMortyDetailContainer extends Component {
  state = {
    character: {},
  }

  async componentDidMount() {
    const { id } = this.props.match.params;
    const character = await findCharacterById();

    this.setState({
      character,
    });
  }

  render() {
    const { character } = this.state;
    return <CharacterDetail { ...character } />;
  }
}
