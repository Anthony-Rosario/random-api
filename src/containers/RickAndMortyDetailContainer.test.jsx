import React from 'react'
import { render, screen } from '@testing-library/react'
import RickAndMortyDetailContainer from './RickAndMortyDetailContainer'
import { MemoryRouter } from 'react-router-dom'

describe('RickAndMortyDetailContainer', () => {
  it('should link to a page with a single character and its relational data', async() => {
    render(<MemoryRouter><RickAndMortyDetailContainer match={{
      params: {
        id: 1
      }
    }}/></MemoryRouter>);
    const li = await screen.findByRole('list', { name: 'character' });

    expect(li).toMatchSnapshot();
  })
})