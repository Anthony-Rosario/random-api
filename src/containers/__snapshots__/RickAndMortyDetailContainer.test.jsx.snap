// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RickAndMortyDetailContainer should link to a page with a single character and its relational data 1`] = `
<li
  aria-label="character detail"
>
  <a
    href="/"
  >
    Back to the list
  </a>
  <img />
  <p />
  <p />
  <p />
</li>
`;
